model/task/generate_tasks_estimations.go

```diff
@@ -0,0 +1,72 @@
+package task
+
+import (
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/utility"
+	adb "github.com/mongodb/anser/db"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	numRevisionsToSearch = 10
+)
+
+func (t *Task) setGenerateTasksEstimations() error {
+	if !t.GenerateTask || (t.EstimatedNumGeneratedTasks != nil && t.EstimatedNumActivatedGeneratedTasks != nil) {
```

Comment:
`t.GenerateTask` will be false if the generate.tasks call isn't in the main task command block. So if generate.tasks is called in `pre`, `post`, `timeout` etc, it won't get an estimated number of tasks. Is that okay?

Reply:
I added `GeneratedTasks` to the pipeline so it doesn't rely on generate task

my only concern is that we'll be running this on every task verses just running it on tasks that have generateTask as true. I feel like it might be wasteful to run this aggregation on every task because a large portion of the tasks would return nothing

Reply:
After thinking about it, I think that we should use t.GenerateTask to see if we should run the estimation algo not. I don't think there are projects that heavily rely on using generate in anything other than the main task block and it isn't worth aggregating on every task just to find the few that may or may not use it.


Also updated the match to only look for successful tasks as failed tasks with 0 generated tasks may make the estimate innacurrate

Reply:
Seems fine with me!